class-tgm-plugin-activation.php

```diff
@@ -630,6 +632,73 @@ protected function do_plugin_install() {
 		}
 
 		/**
+		 * Adjust the plugin directory name if necessary.
+		 *
+		 * The final destination directory of a plugin is based on the subdirectory name found in the
+		 * (un)zipped source. In some cases - most notably GitHub repository plugin downloads -, this
+		 * subdirectory name is not the same as the expected slug and the plugin will not be recognized
+		 * as installed. This is fixed by adjusting the temporary unzipped source subdirectory name to
+		 * the expected plugin slug.
+		 *
+		 * @param string       $source        Path to upgrade/zipfilename.tmp/subdirectory/
+		 * @param string       $remote_source Path to upgrade/zipfilename.tmp
+		 * @param \WP_Upgrader $upgrader      Instance of the upgrader which installs the plugin
+		 *
+		 * @return string $source
+		 */
+		public function maybe_adjust_source_dir( $source, $remote_source, $upgrader ) {
+
+			if ( ! $this->is_tgmpa_page() ) {
+				return $source;
+			}
+
+			// Check for single file plugins
+			$source_files = array_keys( $GLOBALS['wp_filesystem']->dirlist( $remote_source ) );
+			if ( 1 === count( $source_files ) && false === $wp_filesystem->is_dir( $source ) ) {
```

Comment:
Does there need to be a `global $wp_filesystem`, or is it already effectively done so by the `$GLOBALS['wp_filesystem`] on the previous line?